Common/src/matrix_structure.cpp

```diff
@@ -1656,9 +886,9 @@ void CSysMatrix<ScalarType>::BuildILUPreconditioner(bool transposed) {
            upper triangular part, then multiply and modify the matrix.
            Here, Aik' = Aik - Aij*inv(Ajj)*Ajk. ---*/
           
-          if ((kPoint >= jPoint) && (jPoint < (long)nPointDomain)) {
+          if (kPoint > jPoint) {
             
```

Comment:
This was not a bug, but we overwrite the ij block completely so it is not necessary to update it here.